docs/sphinx/developer/components/branchPred.rst

```diff
@@ -35,3 +35,20 @@ Return Address Stack (RAS)
 
 Static Prediction
     Based on the chosen static prediction method of "always taken" or "always not taken", the n-bit saturating counter value in the initial entries of the BTB structure are filled with the weakest variant of taken or not-taken respectively.
+
+Perceptron Predictor
+--------------------
+The ``PerceptronPredictor`` has the same overall structure as the ``GenericPredictor`` but replaces the saturating counter as a means for direction prediction with a perceptron.  The ``PerceptronPredictor`` contains the following logic.
+
+Global History
+    For indexing relevant prediction structures and for retrieving a direction from the perceptrons, a global history can be utilised. The global history value uses n-bits to store the n most recent branch direction outcomes, with the left-most bit being the oldest.
+
+Branch Target Buffer (BTB)
+    For each entry, the BTB stores the most recent target along with a perceptron for an associated direction. The indexing of this structure uses the lower, non-zero bits of an instruction address XOR'ed with the current global branch history value.
+
+    The direction prediction is obtained from the perceptron by taking its dot-product with the global history.  The prediction is not taken if this is negative, or taken otherwise.  The perceptron is updated when its prediction is wrong or when the magnitude of the dot-product is below a pre-determined threshold (i.e., the confidence of the prediction is low).  To update, each ith weight of the perceptron is incremented if the actual outcome of the branch is the same as the ith bit of ``globalHistory_``, and decremented otherwise.
+
+    If the supplied branch type is ``Unconditional``, then the predicted direction is overridden to be taken. If the supplied branch type is ``Conditional`` and the predicted direction is not taken, then the predicted target is overridden to be the next sequential instruction.
```

Comment:
I think "the predicted target is _overridden_ to be the next sequential instruction" is an odd wording. Would something along the lines of "set" be more appropriate as this is precisely the intended behaviour of a BP.

Reply:
I've used the verbatim language from the Generic Predictor documentation.  While I agree that this doesn't mean it is correct, per se, I think that in this case it is good.  Overridden emphasises that not only is the target set, but it replaces the target value that was already there.